### PR TITLE
anvio-minimal v6

### PR DESCRIPTION
**Pull request by ozcan**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.

### CODE REVIEW

File: recipes/anvio-minimal/meta.yaml [hunk 1]
```diff
@@ -15,40 +15,34 @@ build:
...
 requirements:
-  build:
-    - {{ compiler('c') }}
   host:
-    - python >=3,<3.7
-    - pip
-    - numpy
-    - cython
-    - gsl
+    - python ==3.6
```
Comment (bgruening) [inline recipes/anvio-minimal/meta.yaml]: @ozcan all these strict pins are not recommended. Can you loosen them? Python 3.7 is probably also supported isn't it?

File: recipes/anvio-minimal/meta.yaml [hunk 2]
```diff
@@ -15,40 +15,34 @@ build:
...
+    - numpy ==1.17.1
```
Comment (bgruening) [inline recipes/anvio-minimal/meta.yaml]: I would also loosen the numpy version.

File: recipes/anvio-minimal/meta.yaml [hunk 3]
```diff
@@ -15,40 +15,34 @@ build:
...
+    - setuptools ==41.2.0
+    - wheel ==0.33.6
```
Comment (bgruening) [inline recipes/anvio-minimal/meta.yaml]: why do you need wheel and why in this exact version?

File: recipes/anvio-minimal/meta.yaml [hunk 4]
```diff
@@ -59,5 +53,10 @@ about:
   home: http://merenlab.org/software/anvio/index.html
   license: GPL-3.0-or-later
   license_family: GPL3
+  license_file: LICENSE.txt
   summary: "An interactive analysis and visualization platform for omics data"
   dev_url:  https://github.com/merenlab/anvio
+
+extra:
+  skip-lints:
+    - should_be_noarch_python
```
Comment (bgruening) [inline recipes/anvio-minimal/meta.yaml]: If we don't have a compiler, it seems you have a noarch version. Is that not the case?

File: recipes/anvio-minimal/meta.yaml [hunk 5]
```diff
@@ -12,43 +12,38 @@ source:
...
+    - setuptools ==41.2.0
+    - wheel ==0.33.6
+    - pip ==19.2.3
   run:
-    - python >=3,<3.7
-    - numpy
+    - python
+    - {{ pin_compatible('numpy') }}
     - bottle ==0.12.13
     - pysam ==0.15.2
```
Comment (dpryan79) [inline recipes/anvio-minimal/meta.yaml]: Do the version pinnings really have to be so exact?

File: recipes/anvio-minimal/meta.yaml [hunk 6]
```diff
@@ -12,43 +12,38 @@ source:
...
+  skip: True  # [not py36]
   script: python -m pip install --no-deps --ignore-installed .
 
 requirements:
-  build:
-    - {{ compiler('c') }}
   host:
-    - python >=3,<3.7
-    - pip
+    - python
```
Comment (dpryan79) [inline recipes/anvio-minimal/meta.yaml]: Does this still not work with python 3.7?

File: recipes/anvio-minimal/meta.yaml (hunk 5 above)
Comment (ozcan) [reply recipes/anvio-minimal/meta.yaml]: Many of our dependencies have a similar subset of dependencies like `scipy`, `numpy`, `pandas`, which causes some long conflict chains, especially on `Python 3.7`. We found suggesting users create a new environment with frozen dependencies on `Python 3.6` works best under these circumstances, for now. We are planning to loosen dependencies for the next release and hoping to target `Python 3.7` too. We still have limited experience in distributing on Conda and any suggestions and tips are highly appreciated.

File: recipes/anvio-minimal/meta.yaml (hunk 1 above)
Comment (ozcan) [reply recipes/anvio-minimal/meta.yaml]: Hi thank you for your review, I tried to answer version pinning here:
https://github.com/bioconda/bioconda-recipes/pull/17994#discussion_r333356134

File: recipes/anvio-minimal/meta.yaml (hunk 3 above)
Comment (ozcan) [reply recipes/anvio-minimal/meta.yaml]: The version number does not have to be exact but we pinned it as well with others. I don't know exactly what depends on it but I remember seeing `wheel` many times in `conda-build` conflict logs.

File: recipes/anvio-minimal/meta.yaml (hunk 6 above)
Comment (ozcan) [reply recipes/anvio-minimal/meta.yaml]: Unfortunately not, I tried with the newest versions of our dependencies, some does not support Python 3.7. Expectedly some older versions also do not work on 3.7. So we had to manually change versions to find a working combination. We pinned the first combination we got it to work.

Comment (bgruening) [reply recipes/anvio-minimal/meta.yaml]: Do you recall which of you dependencies do not support 3.7. I guess the package is just missing for 3.7 and we can fix this package instead.

Comment (dpryan79) [reply recipes/anvio-minimal/meta.yaml]: I'll tweak the recipe and have it test for the python 3.7 issue.

**Comment by dpryan79** [comment]
Oh, it's the strict snakeMake pinning that's a problem. That old version predated python 3.7.

**Comment by ozcan** [comment]
Thank you for looking into this @dpryan79. It seems we directly import and use `snakemake` here: https://github.com/merenlab/anvio/blob/master/anvio/workflows/__init__.py#L164
I am kindly pinging @ShaiberAlon here to get his feedback on this subject.

**Comment by ShaiberAlon** [comment]
Is the issue here the fact that we require an older version of snakemake?

We indeed use the snakemake API, as shown in the code line @ozcan cited.

We could test our workflows with the latest version of snakemake and if all is well then we can require the latest version.

Please let me know if I understood the issue correctly.

**Comment by dpryan79** [comment]
@ShaiberAlon Yes, that's correct.

**Comment by meren** [comment]
Thank you very much for your time and help with this, @dpryan79!

**Comment by dpryan79** [comment]
No problem, let me know if you end up needing help loosening up version restrictions for the next release.

**Comment by meren** [comment]
Next time we will not leave everything to the last minute and be very happy to have your input. Thank you very much.

**Pull request merged**